docs/release.md

```diff
@@ -0,0 +1,73 @@
+# Releasing RukPkak
+
+This guide outlines the release process for the rukpak project.
+
+## Overview
+
+The rukpak project uses [GoReleaser](https://goreleaser.com/) to automatically produce multi-arch container images and release artifacts during the release process.
+
+In order to create a new minor version release, simply create a new tag locally, push that tag up to the upstream repository remote, and let automation handle the rest.
```

Comment:
nit: should we mention that the tag should be valid semver (and a valid go module version), and that the project follows semver?